apis/apiextensions/v1/xrd_types.go

```diff
@@ -30,12 +30,16 @@ type CompositeResourceDefinitionSpec struct {
 	// Group specifies the API group of the defined composite resource.
 	// Composite resources are served under `/apis/<group>/...`. Must match the
 	// name of the XRD (in the form `<names.plural>.<group>`).
-	// +immutable
+	// +kubebuilder:validation:Required
```

Comment:
Is there a reason to mark fields as required or optional in this PR? Unless enforcing immutability requires this change, I would suggest keeping the change focused only on immutability.

Reply:
@negz its not necessary to mark fields as required or optional to enforce immutability. I will update this.

Reply:
@negz when the kubebuilder validation is set to required then we enforce immutability upon object creation time and when its set to optional then we enforce immutability after first modification i.e. immutability after its first write. So I think its required for enforcing immutability.

Reply:
@negz I have removed the kubebuilder validation